implementations/rust/ockam/ockam_api/src/nodes/iolets.rs

```diff
@@ -6,40 +6,106 @@ use ockam_core::compat::borrow::Cow;
 #[cfg(feature = "tag")]
 use ockam_core::TypeTag;
 
-/// Request body to create an inlet
-#[derive(Debug, Clone, Decode, Encode)]
+/// Distinguish between inlet and outlet portals in the API
+#[derive(Copy, Clone, Debug, Decode, Encode, PartialEq)]
+#[rustfmt::skip]
+#[cbor(index_only)]
+pub enum IoletType {
+    #[n(0)] Inlet,
+    #[n(1)] Outlet,
+}
+
+/// Request body to create an inlet or outlet
+#[derive(Clone, Debug, Decode, Encode)]
 #[rustfmt::skip]
 #[cbor(map)]
-pub struct CreateInlet<'a> {
+pub struct CreateIolet<'a> {
     #[cfg(feature = "tag")]
     #[n(0)]
     tag: TypeTag<1407961>,
-    #[n(1)] pub addr: Cow<'a, str>,
+    /// The type of portal endpoint to create
+    #[n(1)] pub tt: IoletType,
+    /// The address the portal should connect or bind to
+    #[b(2)] pub addr: Cow<'a, str>,
+    /// The forwarding address (must be ockam routing address)
+    ///
+    /// This field will be disregarded for portal outlets.  Portal
+    /// inlets MUST set this value to configure their forwarding
+    /// behaviour.  This can either be the address of an already
+    /// created outlet, or a forwarding mechanism via ockam cloud.
+    #[b(3)] pub fwd: Option<Cow<'a, str>>,
```

Comment:
This name was misleading for me, as we have forwarding addresses. Could this have different name?
Also, what is the format for it? Do we want to use multiaddr for any user-facing Addresses/Routes?